src/ex2/main.js

```diff
@@ -0,0 +1,148 @@
+const validTask = "valid-task-btn"
+const newTasks = ".new-tasks-container"
+
+
+class Main {
+  constructor(addTaskBtn, addTaskInput) {
+    this.addTaskBtn = addTaskBtn;
+    this.addTaskInput = addTaskInput;
+    this.diplayedTasksSet = new Set();
+    this.todoList;
+  }
+
+  init() {
+    this.addTaskInput.addEventListener("input", () => this.onTaskInput());
+
+    this.todoList = new ItemManager(
+      this.addTaskBtn,
+      this.addTaskInput,
+      [],
+      document.querySelector(".clear-all"),
+      {
+        onAddTask: (taskId) => this.onAddTask(taskId),
+        onTaskDelete: (todoToDelete) => this.onTaskDelete(todoToDelete),
+        onCheckUncheckTask: (taskTxt) => this.onCheckUncheckTask(taskTxt),
+        onFinishedAll: () => this.onFinishedAll(),
+        onClearAll: () => this.onClearAll(),
+      }
+    );
+  }
+
+  // Add task will be clickable only for a non empty/"only spaces" task
+  onTaskInput() {
+    if (!/^\s*$/.test(this.addTaskInput.value)) {
+      this.addTaskBtn.classList.add(validTask);
+    } else if (/^\s*$/.test(this.addTaskInput.value)) {
+      this.addTaskBtn.classList.remove(validTask);
+    }
+  }
+
+  clearAddTaskInput() {
+    this.addTaskInput.value = "";
+    this.addTaskBtn.classList.remove(validTask);
+  }
+
+  onAddTask(taskId) {
+    // Render tasks
+    this.todoList.todos.forEach((todo) => {
+      if (!this.diplayedTasksSet.has(todo.id)) {
+        document
+          .querySelector(newTasks)
+          .appendChild(this.createTask(todo));
+        this.diplayedTasksSet.add(todo.id);
+      }
+    });
+
+    this.clearAddTaskInput();
+    this.updateTasksLeft();
+
+    // list is empty and first task was added -> hide 'Done and Done' and show footer
+    if (this.todoList.todos.length === 1) {
+      this.toggleFinishedAllAndFooter();
+    }
+
+    const input = document.querySelector(`#task-${taskId}`);
+    input.addEventListener("click", (event) => this.onTaskPress(event, taskId));
+  }
+
+  onCheckUncheckTask(taskTxtElem) {
+    this.updateTasksLeft();
+    // Crossing/uncrossing a done task's text
+    taskTxtElem.classList.toggle("done-task-txt");
+  }
+
+  onTaskPress(event, taskId) {
+    const isTaskExist = document.querySelector(`#task-${taskId}`);
+    if (isTaskExist) {
+      const wasClickedOnTaskValue =
+        !document
+          .querySelector(`#task-checkbox-${taskId}`)
+          .contains(event.target) &&
+        !document
+          .querySelector(`#trash-button-${taskId}`)
+          .contains(event.target);
+      if (wasClickedOnTaskValue) {
+        alert(document.querySelector(`#task-txt-${taskId}`).innerText);
+      }
+    }
+  }
+
+  onTaskDelete(todoToDelete) {
+    document.querySelector(`#task-${todoToDelete.id}`).remove();
+    this.diplayedTasksSet.delete(todoToDelete.id);
+    this.updateTasksLeft();
+  }
+
+  onClearAll() {
+    const nodes = document.querySelector(newTasks).childNodes;
```

Comment:
if you would render the todos from itemManager you could make your solution simpler (just assign empty array to todos and render it)